### PR TITLE
Fix Flake8 violations for files in the test/ directory

### PR DESCRIPTION
**Pull request by LucyJimenez**
xref https://github.com/airspeed-velocity/asv/issues/1042

Fix pep8 issues on:
- `test_util.py`
- `test_subprocess.py`

### CODE REVIEW

File: test/test_util.py [hunk 1]
```diff
@@ -1,18 +1,15 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-import io
-import locale
+import locale # noqa F401 imported but unused
```
Comment (LucyJimenez) [inline test/test_util.py:2]: The `locale` library is not used and in the code of the file it is not found anywhere else, but when I remove it I get an error in [AppVeyor CI]( https://ci.appveyor.com/project/airspeed-velocity/asv/builds/42784236). Any idea why this happens?

File: test/test_subprocess.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def test_timeout():
 sys.stdout.flush()
 sys.stderr.flush()
 subprocess.call([sys.executable, "-c",
-    "import sys, subprocess; subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"])
+    "import sys, subprocess; subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"])  # noqa E501 line too long
```
Comment (datapythonista) [inline test/test_subprocess.py:37]: I don't see any reason why this can't be splitted in multiple lines.

File: test/test_util.py (hunk 1 above)
Comment (datapythonista) [reply test/test_util.py:2]: Can you move this to it's own PR, so we can isolate and see the CI failure without other changes.

**Comment by LucyJimenez** [comment]
@datapythonista I'm going to slip this PR to get more info about the errors.

**Comment by datapythonista** [comment]
> @datapythonista I'm going to slip this PR to get more info about the errors.

Not sure what you mean, but the line too long just needs to be split and for the `locale` import, you can just have a message that things fail if removed, and create and link an issue to check why.

File: test/test_subprocess.py (hunk 2 above)
Comment (LucyJimenez) [reply test/test_subprocess.py]: I'm not sure why it is not easy to split this line. I tried a couple of things, and it doesn’t work:
Option 1:
image
Option 2:
image

Could you please help me?

Comment (datapythonista) [reply test/test_subprocess.py]: This is Python code as a string. It needs to be valid Python code when converted to a `.py` file. And this is clearly not, as you have a multiline string with single quotes.

File: test/test_subprocess.py [hunk 3]
```diff
@@ -34,7 +34,8 @@ def test_timeout():
 sys.stdout.flush()
 sys.stderr.flush()
 subprocess.call([sys.executable, "-c",
-    "import sys, subprocess; subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"])
+    "import sys, subprocess; subprocess.call([sys.executable, '-c', \
+    'import time; time.sleep(360)'])"])
```
Comment (datapythonista) [inline test/test_subprocess.py:38]: This is quite hard to read. The code itself is easy to read, we should split it in a way that makes things more clear.

I think I'd have the parameters of `call` in their own line: `sys.executable, '-c', 'import time; time.sleep(360)'`.

And I would avoid using a backslash, as this makes things harder to read, besides introducing spaces. I'd just have one string per line, which Python will automatically concatenate.

**Comment by LucyJimenez** [comment]
@datapythonista  The CI / test (pypy-3.8) has been running for ~50 minutes. Is it normal?

image

**Review by datapythonista — COMMENTED** [review]
I will have a look at the pypy build if it happens again.

File: test/test_subprocess.py [hunk 4]
```diff
@@ -33,8 +33,8 @@ def test_timeout():
 sys.stderr.write("Stderr before waiting\n")
 sys.stdout.flush()
 sys.stderr.flush()
-subprocess.call([sys.executable, "-c",
-    "import sys, subprocess; subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"])
+subprocess.call([sys.executable, "-c", "import sys, subprocess;"
+"subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"])
```
[the remark this answers is not in the document]
Comment (LucyJimenez) [reply test/test_subprocess.py]: Thank you, I'll put it in the code.

[the remark this answers is not in the document]
Comment (LucyJimenez) [reply test/test_subprocess.py]: > ```python
> subprocess.call([
>     sys.executable,
>     "-c",
>     "import sys, subprocess; subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"
> ])
> ```

File: test/test_subprocess.py [hunk 5]
```diff
@@ -33,8 +33,12 @@ def test_timeout():
 sys.stderr.write("Stderr before waiting\n")
 sys.stdout.flush()
 sys.stderr.flush()
-subprocess.call([sys.executable, "-c",
-    "import sys, subprocess; subprocess.call([sys.executable, '-c', 'import time; time.sleep(360)'])"])
+subprocess.call([
+    sys.executable,
+    "-c",
+    "import sys, subprocess; subprocess.call("
+    "[sys.executable, -c', import time; time.sleep(360)'])"
```
Comment (datapythonista) [inline test/test_subprocess.py:40]:
```suggestion
    "[sys.executable, '-c', import time; time.sleep(360)'])"
```
A quote is missing

Comment (LucyJimenez) [reply test/test_subprocess.py]: @datapythonista I fix the quote missing, but it is still not working.  Do you think I could use this code instead?

```
subprocess.call([sys.executable, "-c",
    "import sys, subprocess; subprocess.call(\
    [sys.executable, '-c', 'import time; time.sleep(360)'])"])
```

Comment (datapythonista) [reply test/test_subprocess.py]: The backslash is irrelevant, and as I mentioned, a discouraged practice in Python. There is another quote missing:
```python
[sys.executable, '-c', import time; time.sleep(360)']
```

```python
[sys.executable, '-c', 'import time; time.sleep(360)']
```

File: test/test_util.py [hunk 6]
```diff
@@ -1,18 +1,15 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-import io
-import locale
+import locale # noqa F401 imported but unused see #1106
```
Comment (datapythonista) [inline test/test_util.py:2]: The errors in #1106 seem unrelated to this. Can you try to remove this again, and see if the error is relevant this time please.

**Review by datapythonista — COMMENTED** [review]
There are conflicts that need to be solved, and one comment.

File: test/test_util.py (hunk 6 above)
Comment (LucyJimenez) [reply test/test_util.py]: I removed the `import locale` and there are errors for the [CI / test (ubuntu-latest, pypy-3.7) (pull_request) ](https://github.com/airspeed-velocity/asv/runs/5489967552?check_suite_focus=true) and the [CI / test (ubuntu-latest, pypy-3.8) (pull_request)](https://github.com/airspeed-velocity/asv/runs/5489967597?check_suite_focus=true). The tests are passed for the `test/test_subprocess.py` and `test/test_util.py` files in both cases.

image

image

Comment (datapythonista) [reply test/test_util.py]: These failures seem unrelated. Can you just remove the `locale` import from this PR.

Comment (LucyJimenez) [reply test/test_util.py]: Done!

**Comment by LucyJimenez** [comment]
@datapythonista, I think now this PR is ready for the last review :)

**Comment by datapythonista** [comment]
Thanks @LucyJimenez
